### PR TITLE
Feature/hw 66662 manage transfer method

### PR DESCRIPTION
**Pull request by shanavas123**

### CODE REVIEW

File: transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/ListTransferMethodFragment.java [hunk 1]
```diff
@@ -361,8 +362,18 @@ void bind(@NonNull final TransferMethod transferMethod) {
                 Locale locale = new Locale.Builder().setRegion(
                         transferMethod.getField(TRANSFER_METHOD_COUNTRY)).build();
                 mIcon.setText(getStringFontIcon(mIcon.getContext(), type));
-                mTransferMethodCountry.setText(locale.getDisplayName());
-                mTransferMethodIdentification.setText(transferMethodIdentification);
+                if(type.equals(PREPAID_CARD))
+                {
```
Comment (peter-joseph) [inline transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/ListTransferMethodFragment.java:366]: Open `{` needs to be in line 365, please follow the current code convention

**Review by peter-joseph — COMMENTED** [review]
Please following coding guidelines in this doc https://confluence.site1.hyperwallet.local/display/~bolynyk/Android+SDK+Architecture

I also slacked in the channel

File: commonui/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -26,6 +28,7 @@
     <string name="cash_pickup_mg_font_icon">\ue905</string>
 
     <string name="endingIn">ending in %s</string>
+    <string name="card_brand_with_four_digits">%1s \u00b7\u00b7\u00b7\u00b7 %2s</string>
```
[the remark this answers is not in the document]
Comment (shanavas123) [reply commonui/src/main/res/values/strings.xml]: fixed

File: transfermethodui/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -76,4 +76,5 @@
     <string name="mobileTransferMethodsHeader">Transfer methods</string>
 
     <string name="mobileRemoveEAconfirm">Remove transfer method</string>
+    <string name="prepaid_card_management_info">"Log in using a web browser to manage your card"</string>
```
[the remark this answers is not in the document]
Comment (shanavas123) [reply transfermethodui/src/main/res/values/strings.xml]: fixed

File: commonui/src/main/java/com/hyperwallet/android/ui/common/view/TransferMethodUtils.java [hunk 4]
```diff
@@ -181,4 +185,21 @@ private static String getFourDigitsIdentification(@NonNull final Context context
 
         return context.getString(stringResId, identificationText);
     }
+
+    private static String getFourDigitsIdentificationWithCardBrand(@NonNull final Context context,
+                                                      @NonNull final TransferMethod transferMethod,
+                                                      @NonNull @TransferMethod.TransferMethodFieldKey final String card_no_fieldKey,
+                                                                  @NonNull @TransferMethod.TransferMethodFieldKey final String card_brand_fieldKey,
+                                                      @StringRes final int stringResId) {
```
[the remark this answers is not in the document]
Comment (shanavas123) [reply commonui/src/main/java/com/hyperwallet/android/ui/common/view/TransferMethodUtils.java]: fixed

File: transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/ListTransferMethodFragment.java (hunk 1 above)
Comment (shanavas123) [reply transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/ListTransferMethodFragment.java]: fixed

File: transfermethodui/src/androidTest/java/com/hyperwallet/android/ui/transfermethod/ListTransferMethodTest.java [hunk 5]
```diff
@@ -520,6 +518,11 @@ private String getEndingIn(String ending) {
 
     }
 
+    private String getCardBrandWithFourDigits(String cardBrand,String ending) {
+        return cardBrand+"\u0020\u2022\u2022\u2022\u2022\u0020"+ending;
+
+    }
```
Comment (vwagh-hw) [inline transfermethodui/src/androidTest/java/com/hyperwallet/android/ui/transfermethod/ListTransferMethodTest.java:524]: 1. There should be space between parameters after comma 
2. and there should be space around `+` sign
3. `ending` should be named as `endingDigits`
4. Remove Unnecessary empty line after return

Comment (shanavas123) [reply transfermethodui/src/androidTest/java/com/hyperwallet/android/ui/transfermethod/ListTransferMethodTest.java]: fixed

**Review by rajkumar-balakrishnasamy — DISMISSED** [review]

File: commonui/src/main/java/com/hyperwallet/android/ui/common/view/TransferMethodUtils.java [hunk 6]
```diff
@@ -181,4 +184,21 @@ private static String getFourDigitsIdentification(@NonNull final Context context
 
         return context.getString(stringResId, identificationText);
     }
+
+    private static String getFourDigitsIdentificationWithCardBrand(@NonNull final Context context,
+            @NonNull final TransferMethod transferMethod,
+            @NonNull @TransferMethod.TransferMethodFieldKey final String cardNumberFieldKey,
+            @NonNull @TransferMethod.TransferMethodFieldKey final String cardBrandFieldKey) {
```
Comment (peter-joseph) [inline commonui/src/main/java/com/hyperwallet/android/ui/common/view/TransferMethodUtils.java:191]: `cardNumberFieldKey` and `cardBrandFieldKey` parameter values are from static constant, why is it being passed as a parameter? couldn't we just use the constant value directly without passing it as a parameter?

Comment (peter-joseph) [reply commonui/src/main/java/com/hyperwallet/android/ui/common/view/TransferMethodUtils.java:191]: effectively using the following ?
```
transferMethod.getField(CARD_NUMBER);
transferMethod.getField(CARD_BRAND);
```

Comment (shanavas123) [reply commonui/src/main/java/com/hyperwallet/android/ui/common/view/TransferMethodUtils.java]: fixed.

**Review by peter-joseph — DISMISSED** [review]
Looks good to me now, thanks 👍

**Review by rajkumar-balakrishnasamy — APPROVED** [review]

**Review by vwagh-hw — APPROVED** [review]

**Review by jsaini-hw — APPROVED** [review]

**Review by vwagh-hw — APPROVED** [review]

**Pull request merged**